promql/parse.go

```diff
@@ -1074,13 +1074,19 @@ func (p *parser) checkType(node Node) (typ model.ValueType) {
 
 	case *Call:
 		nargs := len(n.Func.ArgTypes)
-		if na := nargs - n.Func.OptionalArgs; na > len(n.Args) {
+		if na := nargs - n.Func.Variadic; n.Func.Variadic >= 0 && na > len(n.Args) {
```

Comment:
@fabxc: You said if Variadic >= 0, the function should contain either 0 of those optional arguments or all of those arguments. Hence, anything between 0 and the optional args is unacceptable right?